appsignals-tests/contract-tests/src/test/java/software/amazon/opentelemetry/appsignals/test/jdbc/JdbcMySQLTest.java

```diff
@@ -48,11 +52,16 @@ public void afterEach() {
     mySQLContainer.stop();
   }
 
-  @Test
-  public void testSuccess() {
+  private static Stream<DBOperation> dbOperations() {
+    return Stream.of(DBOperation.SELECT, DBOperation.CREATE_DATABASE);
+  }
+
+  @ParameterizedTest
+  @MethodSource("dbOperations")
+  public void testSuccess(DBOperation operation) {
     assertSuccess(
```

Comment:
Here you see the main problem IMHO.

You are forced to bend the usage of `assertSuccess` for each operation, note how `assertSuccess` comes from the super class `JdbcContractTestBase`

I would ask myself what is the responsability of `JdbcContractTestBase.assertSuccess`? And I would say that such method should:

1. Not be the one making http requests nor collecting traces
2. It should just check that the requests got back is around 200
3. Check the few traces and span that are generic for each JDBC driver

(I would honestly remove the check for 200, and just pass around the status code you expect as parameter.)

At this point in the other drivers, you can build on top of them.

All the common checks are already in the Jdbc, and in the MySQL driver you can check only the traces and span that depends on the MySQL driver. Similarly for PG and H2.